editor/store/reducer.js

```diff
@@ -122,6 +122,30 @@ function shouldOverwriteState( action, previousAction ) {
 	return false;
 }
 
+/**
+ * Higher-order reducer targeting the combined editor reducer, augmenting
+ * block UIDs in remove action to include cascade of inner blocks.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+const withInnerBlocksRemoveCascade = ( reducer ) => ( state, action ) => {
```

Comment:
This makes me wonder if we should handle the `editor.blockOrder` and `editor.blockUids` in the same reducer without "combineReducer".

But I can be on board with this idea of HoR.

Reply:
In this particular case, it's kinda nice that we can augment the action without changing any of the original behavior of the reducer. Basically it's something of a small middleware which adds more UIDs to the blocks being removed.

But you raise an interesting point. Most operations which impact blocks impact both the ordering and the source values, and it could imagine it would be useful to have access to both in the process of doing so. Since we don't desperately need it right now, I'd be inclined to tackle this separately. Would be curious if it might simplify the code.